adala/skills/collection/classification.py

```diff
@@ -70,9 +110,8 @@ class ClassificationSkill(TransformSkill):
     def validate_response_model(self):
 
         if self.response_model:
-            raise NotImplementedError(
-                "Classification skill does not support custom response model yet."
-            )
+            # NOTE: the field_schemas provided are not the same as response_model.schema() (json schemas don't roundtrip), so can't validate here
```

Comment:
Do we really want to bypass the other validations in this case? Definitely might be missing the context of when we get here vs not...

Reply:
the validations in this function are only on the field_schema or output_template used to construct the response_model, not on the content of the resulting response_model. I added a test that shows how to pass a response_model directly that's identical to the one constructed. Basically we assume that if you pass this field you know what you're doing, since there already wasn't validation for it :)

Reply:
could write an equivalent of `validate_schema` for the response_model.schema() but unfortunately can't use the existing one